packages/liveblocks-core/src/room.ts

```diff
@@ -2005,11 +2018,39 @@ export function createRoom<
   let _getStorage$: Promise<void> | null = null;
   let _resolveStoragePromise: (() => void) | null = null;
 
+  function processInitialStorage(message: InitialDocumentStateServerMsg) {
+    const unacknowledgedOps = new Map(context.unacknowledgedOps);
+    createOrUpdateRootFromMessage(message, doNotBatchUpdates);
+    applyAndSendOps(unacknowledgedOps, doNotBatchUpdates);
+    _resolveStoragePromise?.();
+    notifyStorageStatus();
+    eventHub.storageDidLoad.notify();
+  }
+
+  async function streamStorage() {
+    if (!managedSocket.authValue) {
+      return;
+    }
+    const result = await streamFetch(
+      managedSocket.authValue.type === "public"
+        ? managedSocket.authValue.publicApiKey
+        : managedSocket.authValue.token.raw,
+      config.roomId
+    );
+    const items = (await result.json()) as IdTuple<SerializedCrdt>[];
+    processInitialStorage({ type: ServerMsgCode.INITIAL_STORAGE_STATE, items });
```

Comment:
To be ahead of race conditions later, when this code grows more complicated, we may want to also think about cancelling any in-flight HTTP requests, or otherwise re-validating its mandate when its response arrives in the client.

Areas where this might get tricky in the future as this code grows more complex is when disconnecting/reconnecting a client while such an HTTP request is under way. In theory, you can send the HTTP request, disconnect/reconnect, send the HTTP request again, and the HTTP requests arriving back out of order, or multiple times, which may lead to unforeseen edge cases.